benchmarks/MOABB/dataio/preprocessing.py

```diff
@@ -24,39 +24,82 @@ def to_tensor(epoch):
 cached_create_filter = cache(mne.filter.create_filter)
 
 
-@takes("epoch", "info", "target_sfreq", "fmin", "fmax")
-@provides("epoch", "sfreq", "target_sfreq", "fmin", "fmax")
-def bandpass_resample(epoch, info, target_sfreq, fmin, fmax):
-    """Bandpass filter and resample an epoch."""
-
-    bandpass = cached_create_filter(
-        None,
-        info["sfreq"],
-        l_freq=fmin,
-        h_freq=fmax,
-        method="fir",
-        fir_design="firwin",
-        verbose=False,
-    )
-
-    # Check that filter length is reasonable
-    filter_length = len(bandpass)
-    len_x = epoch.shape[-1]
-    if filter_length > len_x:
-        # TODO: These long filters result in massive performance degradation... Do we
-        #       want to throw an error instead? This usually happens when fmin is used
-        logging.warning(
-            "filter_length (%i) is longer than the signal (%i), "
-            "distortion is likely. Reduce filter length or filter a longer signal.",
-            filter_length,
-            len_x,
+def bandpass_resample():
+    @takes("epoch", "info", "target_sfreq", "fmin", "fmax")
+    @provides("epoch", "sfreq", "target_sfreq", "fmin", "fmax")
+    def _bandpass_resample(epoch, info, target_sfreq, fmin, fmax):
+        """Bandpass filter and resample an epoch."""
```

Comment:
This is more what I had in mind as it makes it easy to configure from the hparams file. See how it is a function which creates a dynamic item with the desired parameters?

DONT commit the suggestion directly, as it is just to illustrate

```suggestion
def bandpass_resample(target_sfreq, fmin, fmax):
    @takes("epoch", "info")
    @provides("epoch")
    def _bandpass_resample(epoch, info, target_sfreq, fmin, fmax):
        """Bandpass filter and resample an epoch."""
```

Reply:
This will allow us to do something like this in the yaml file:
```yaml
dynamic_items:
- !apply:bandpass_resample
  fmin: 0.5
  fmax: 22
  target_sfreq: 128
```